core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -184,6 +201,78 @@ private void error(PrintWriter out, String msg) throws IOException {
     }
 
     /**
+     * This extension provides a Ping protocol that allows people to verify that the TcpSlaveAgentListener is alive.
+     * We also use this
+     * @since
+     */
+    @Extension
+    public static class PingAgentProtocol extends AgentProtocol {
+
+        private final byte[] ping;
+
+        public PingAgentProtocol() {
+            try {
+                ping = "Ping\n".getBytes("UTF-8");
```

Comment:
I would like to introduce you to `String.getBytes(Charset)` and `StandardCharsets.html.UTF_8` - they will be your new best friends!

"Ping\n".getBytes([StandardCharsets.UTF_8](http://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html#UTF_8))